### DISCUSSION TITLE
Auto created firewall rule poses security risk

### DISCUSSION THREAD

**Discussion by olfek** [nodejs/help#4720] [Q&A] [answered]
Today I discovered something a bit alarming.

`Node.js JavaScript Runtime` is an **inbound** firewall rule created automatically by the Node JS installer on Windows.

It looks like this:

```
Name: Node.js JavaScript Runtime
Group: -
Profile: Private, Public
Enabled: Yes
Action: Allow
Override: No
Program: C\program files\nodejs\node.exe
Local Address: Any
Remote Address: Any
Protocol: TCP
Local Port: Any
Remote Port: Any
Authorized Users: Any
Authorized Computers: Any
Authorized Local Principals: Any
Local User Owner: Any
Application Package: Any
```

**Comment by olfek** [comment]
I would say this would be OK on the `Private` profile, but, even then, this kind of rule should never be created _without_ the users consent.

**Comment by olfek** [comment]
Pinging you because you are part of the security working group 🙂 @mhdawson

**Comment by bnoordhuis** [comment]
You probably installed node as part of some other package (ex. adobe bundles node with some of their products) because node's own installer doesn't add firewall exceptions. You can check that yourself, the relevant files are in tools/msvs in the repo.

**Reply by olfek** [reply]
I'm 100% sure I used the Windows Node JS installer

**Reply by olfek** [reply]
Maybe it was some NPM CLI tool then?

**Reply by bnoordhuis** [reply]
Could be. Maybe you clicked "allow" when you ran node or npm for the first time.

**Reply by olfek** [reply]
Yeah I think I did, but I didn't know the rule would be so permissive.

The firewall rule request popup, is that _not_ triggered by the installer?

**Reply by bnoordhuis** [reply, marked as answer]
No, that comes from Windows itself.

**Reply by olfek** [reply]
🤦🤦🤦 All these years, I've been clicking "Allow access" thinking the newly installed application requested **necessary** Firewall changes, I'm an idiot.
